python/ray/dag/compiled_dag_node.py

```diff
@@ -903,7 +936,10 @@ def _preprocess(self) -> None:
                     # DAG's input node as the actual upstream node
                     upstream_task = self.idx_to_task[self.input_task_idx]
 
-                elif isinstance(upstream_task.dag_node, InputNode):
+                elif isinstance(upstream_node, InputNode):
+                    if isinstance(dag_node, InputAttributeNode):
```

Comment:
The `args` of `InputAttributeNode` is an empty list, but its `_upstream_nodes` is not empty and should include `InputNode`. If we don't add continue here, a `ValueError` will be raised later.